toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/sceneview/ArcGISSceneViewTableTopActivity.kt

```diff
@@ -0,0 +1,154 @@
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource
+import com.esri.arcgisruntime.mapping.NavigationConstraint
+import com.esri.arcgisruntime.mapping.Surface
+import com.esri.arcgisruntime.mapping.view.Camera
+import com.esri.arcgisruntime.mapping.view.DefaultSceneViewOnTouchListener
+import com.esri.arcgisruntime.toolkit.ar.ArcGISArView
+import com.esri.arcgisruntime.toolkit.extension.logTag
+import com.esri.arcgisruntime.toolkit.test.R
+import kotlinx.android.synthetic.main.activity_arcgissceneview.arcGisArView
+import java.net.URI
+
+class ArcGISSceneViewTableTopActivity : AppCompatActivity(), ArcGISArView.OnStateChangedListener {
+
+    private var _scene: ArcGISScene? = null
+    private var pointCloudLayer: Layer? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_arcgissceneview_tabletop)
+        arcGisArView.registerLifecycle(lifecycle)
+        arcGisArView.addOnStateChangedListener(this)
+
+        arcGisArView.sceneView.setOnTouchListener(object : DefaultSceneViewOnTouchListener(arcGisArView.sceneView) {
+            override fun onSingleTapConfirmed(motionEvent: MotionEvent?): Boolean {
+                motionEvent?.let {
+                    if (_scene != null && pointCloudLayer != null && !_scene!!.operationalLayers.contains(
+                            pointCloudLayer
+                        )
+                    ) {
+                        _scene!!.operationalLayers.add(pointCloudLayer)
+                    }
+                    with(Point(motionEvent.x.toInt(), motionEvent.y.toInt())) {
+                        arcGisArView.setInitialTransformationMatrix(this)
+                    }
+                }
+                return false
+            }
+        })
+    }
+
+    override fun onStateChanged(state: ArcGISArView.ArcGISArViewState) {
+        when (state) {
+            ArcGISArView.ArcGISArViewState.NOT_INITIALIZED -> {
+                //no-op
+            }
+            ArcGISArView.ArcGISArViewState.INITIALIZING -> {
+                // no-op
+            }
+            ArcGISArView.ArcGISArViewState.INITIALIZED -> {
+                arcGisArView.originCamera = Camera(50.94334724, 6.96472093, 44.412, 267.0, 90.0, 0.0)
+                arcGisArView.translationTransformationFactor = 4000.0
+
+                with(arcGisArView.sceneView) {
+                    _scene = ArcGISScene()
+                    scene = _scene
+                    val surface = Surface()
+                    ArcGISTiledElevationSource("http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer").let {
+                        surface.elevationSources.add(it)
+                    }
+                    surface.navigationConstraint = NavigationConstraint.NONE
+                    surface.opacity = 0f
+                    scene.baseSurface = surface
+                    val encodedString =
```

Comment:
Can this name be updated? I assume it was still from when the encoding bug occurred, which is not fixed? How about `pointCloudLayerUrl`?

Reply:
Yeah, I think it can be now.